xds/internal/balancer/lrs/balancer.go

```diff
@@ -212,11 +212,11 @@ func (w *xdsClientWrapper) update(newConfig *LBConfig) error {
 		w.loadWrapper.UpdateClusterAndService(w.clusterName, w.edsServiceName)
 	}
 
-	if w.lrsServerName != newConfig.LoadReportingServerName {
+	if w.lrsServerName == nil || *w.lrsServerName != newConfig.LoadReportingServerName {
```

Comment:
Do you think it makes sense to add a utility type like `Optional`. But without generics, it is going to involve some ugly type assertions.

Reply:
Yes, we need generics.

And, this is not comparing two string pointers.
`lrsServerName == nil` is to check the init case, and we never set back to `nil`.